pkg/datapath/l2responder/l2responder.go

```diff
@@ -0,0 +1,334 @@
+	r := tbl.Reader(db.ReadTxn())
+	iter, err := r.Get(statedb.All)
+	if err != nil {
+		log.WithError(err).Error("Error getting all desired proxy table entries")
+	}
+
+	select {
+	case <-ctx.Done():
+		return 0
+
+	case <-iter.Invalidated():
+		maxRev, err = p.partialReconciliation(maxRevIn)
+		if err != nil {
+			log.WithError(err).Error("Error(s) while partial reconciling l2 responder map")
+		}
+
+		return maxRev
+
+	case <-fullReconciliation:
+
+		// The existing `iter` is the result of a `All` query, so this will return all
+		// entries in the table for full reconciliation.
+		maxRev, err = p.fullReconciliation()
+		if err != nil {
+			log.WithError(err).Error("Error(s) while full reconciling l2 responder map")
+		}
+
+		return maxRev
+	}
+}
+
+func (p *l2ResponderReconciler) partialReconciliation(maxRevIn uint64) (maxRev uint64, err error) {
+	var errs error
+
+	maxRev = maxRevIn
+
+	arMap := p.params.L2ResponderMap
+	tbl := p.params.L2AnnouncementTable
+	db := p.params.StateDB
+	log := p.params.Logger
+	lr := cachingLinkResolver{nl: p.params.NetLink}
+
+	log.Info("l2 announcer table invalidated, performing partial reconciliation")
+
+	// Get all changes since the revision we processes
+	r := tbl.Reader(db.ReadTxn())
+	iter, err := r.LowerBound(statedb.ByRevision(maxRevIn))
+	if err != nil {
+		log.WithError(err).Error("Error getting last changes")
+	}
+
+	// A list of desired entries which have been soft deleted
+	var toDelete []*tables.L2AnnounceEntry
+
+	statedb.ProcessEach(iter, func(e *tables.L2AnnounceEntry) error {
+		if e.Revision > maxRev {
+			maxRev = e.Revision
+		}
+
+		// Ignore IPv6 addresses, L2 is IPv4 only
+		if e.IP.To4() == nil {
+			return nil
+		}
+
+		idx, err := lr.LinkIndex(e.NetworkInterface)
+		if err != nil {
+			errs = errors.Join(errs, fmt.Errorf("link index: %w", err))
+			return nil
+		}
+
+		if e.Deleted {
+			toDelete = append(toDelete, e)
+			err = arMap.Delete(e.IP, uint32(idx))
+			if err != nil {
+				errs = errors.Join(errs, fmt.Errorf("delete %s@%d: %w", e.IP, idx, err))
+			}
+			return nil
+		}
+
+		err = arMap.Create(e.IP, uint32(idx))
+		if err != nil {
+			errs = errors.Join(errs, fmt.Errorf("create %s@%d: %w", e.IP, idx, err))
+		}
+
+		return nil
+	})
+
+	// Hard delete, soft deleted entries
+	if len(toDelete) > 0 {
+		txn := db.WriteTxn()
+		w := tbl.Writer(txn)
+		for _, e := range toDelete {
+			if err = w.Delete(e); err != nil {
+				errs = errors.Join(errs, fmt.Errorf("delete from table: %w", err))
+			}
+		}
+		if err = txn.Commit(); err != nil {
+			errs = errors.Join(errs, fmt.Errorf("commit deletion to table: %w", err))
+		}
+	}
+
+	return maxRev, errs
+}
+
+func (p *l2ResponderReconciler) fullReconciliation() (maxRev uint64, err error) {
+	var errs error
+
+	log := p.params.Logger
+	tbl := p.params.L2AnnouncementTable
+	db := p.params.StateDB
+	arMap := p.params.L2ResponderMap
+	lr := cachingLinkResolver{nl: p.params.NetLink}
+
+	log.Info("l2 announcer table full reconciliation")
```

Comment:
Is this really a useful thing to log at Info? Seems more like Debug to me. Same in partial reconciliation. 

If this would be at info level then I'd log at the end and include some stats about what was done. This by itself seems a bit useless.